### PR TITLE
Feature/brv 216

### PR DESCRIPTION
**Pull request by DanielFCubides**
[IDEA VALIDATION] Zip client

## JIRA Ticket URL
https://merlinapp.atlassian.net/browse/BRV-216

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I did add reviewers to this PR
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I created the unit tests and they are passing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+```go
+import (
+    "github.com/merlinapp/zip-recruiter-go-client/jobs"
+)
+```
+so in your go modules file `go.mod` you should hve the following entry:
+
+```
+require (
+    github.com/merlinapp/zip-recruiter-go-client v0.0.6
```
Comment (daltemen) [inline README.md]:
```suggestion
    github.com/merlinapp/zip-recruiter-go-client <version>
```

File: jobs/client.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+func NewZipClient() *ZipClient {
+	apiKey := os.Getenv("ZIP_RECRUITER_KEY")
+	if len(apiKey) == 0 {
+		panic("ZIP_RECRUITER_KEY env variable not set")
+	}
+	return &ZipClient{BaseUrl: apiKey}
+}
+
+func (z *ZipClient) Get(request ZipRequest) (*ZipResponse, error) {
+	url := fmt.Sprintf("%s?search=%s&locatiion=%s&radius_miles=%s&days_ago=%s&jobs_per_page=%s&page=%s&refined_salary=%s&api_key=%s",
```
Comment (daltemen) [inline jobs/client.go]: typo `locatiion`

Comment (daltemen) [reply jobs/client.go]: please use https://golang.org/pkg/net/url/#example_URL `net/url` package is more readable.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module github.com/merlinapp/zip-recruiter-go-client
+
+go 1.13
+
+require (
+	bou.ke/monkey v1.0.2
```
Comment (daltemen) [inline go.mod]: is necessary monkey ?

File: jobs/fixtures/response.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "status": "succeed",
+  "jobs": [
+    {},
+    {},
+    {}
+  ],
+  "": 3
+}
```
Comment (daltemen) [inline jobs/fixtures/response.json]: it would be nice to have a snapshot of the response.

**Review by daltemen — APPROVED** [review]

**Review by daltemen — COMMENTED** [review]
minor comments

File: README.md (hunk 1 above)
Comment (DanielFCubides) [reply README.md]: done

File: jobs/client.go (hunk 2 above)
Comment (DanielFCubides) [reply jobs/client.go]: done

File: go.mod (hunk 3 above)
Comment (DanielFCubides) [reply go.mod]: nop, i will remove it

File: jobs/fixtures/response.json (hunk 4 above)
Comment (DanielFCubides) [reply jobs/fixtures/response.json]: added

File: README.md [hunk 5]
```diff
@@ -1,3 +1,65 @@
...
+Set the environment variable `ZIP_RECRUITER_KEY`
+
+Using go modules, import
+
+```go
+import (
+    "github.com/merlinapp/zip-recruiter-go-client/jobs"
+)
+```
+so in your go modules file `go.mod` you should hve the following entry:
```
Comment (daltemen) [inline README.md]: typo `hve`

File: jobs/client.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+	queryParams := u.Query()
+	queryParams.Set("search", request.Search)
+	queryParams.Set("location", request.Location)
+	queryParams.Set("radius_miles", strconv.Itoa(int(request.RadiusMiles)))
+	queryParams.Set("days_ago", strconv.Itoa(int(request.DaysAgo)))
+	queryParams.Set("jobs_per_page", strconv.Itoa(int(request.JobsPerPage)))
+	queryParams.Set("page", strconv.Itoa(int(request.Page)))
+	queryParams.Set("refine_by_salary", strconv.Itoa(int(request.RefineSalary)))
+	queryParams.Set("api_key", z.ApiKey)
+	u.RawQuery = queryParams.Encode()
```
Comment (daltemen) [inline jobs/client.go]: this method seems large, please move this code to other private method

File: jobs/fixtures/response.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (daltemen) [inline jobs/fixtures/response.json:1]: 👍

**Review by daltemen — COMMENTED** [review]
minors

File: README.md (hunk 1 above)
Comment (DanielFCubides) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (DanielFCubides) [reply README.md]: done

File: jobs/client.go (hunk 6 above)
Comment (DanielFCubides) [reply jobs/client.go]: done

Comment (DanielFCubides) [reply jobs/client.go]: done

**Review by daltemen — APPROVED** [review]

File: jobs/client.go [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+func NewZipClient() Client {
+	apiKey := os.Getenv("ZIP_RECRUITER_KEY")
+	if len(apiKey) == 0 {
+		panic("ZIP_RECRUITER_KEY env variable not set")
+	}
+	baseUrl := os.Getenv("ZIP_RECRUITER_BASE_URL")
+	if len(baseUrl) == 0 {
+		panic("ZIP_RECRUITER_BASE_URL env variable not set")
```
Comment (E-ricus) [inline jobs/client.go]: It is not the best idea to add these as env variables, maybe is better to receive it on the constructor, and the user of the library load it on their envs

**Review by E-ricus — COMMENTED** [review]
Comment applies for all envs

File: jobs/client.go (hunk 8 above)
Comment (DanielFCubides) [reply jobs/client.go]: Done

**Review by daltemen — APPROVED** [review]

**Review by daltemen — APPROVED** [review]

File: jobs/client.go [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+	"net/http"
+	urlUtils "net/url"
+	"strconv"
+)
+
+const (
+	parserError        = "could not parse error"
+	unauthorizedAccess = "unauthorized access"
+	urlParseError      = "could not parse base url"
+)
```
Comment (cristianchaparroa) [inline jobs/client.go]: could you put the errors onto `errors.go` file

**Review by cristianchaparroa — CHANGES_REQUESTED** [review]

File: jobs/client.go (hunk 9 above)
Comment (DanielFCubides) [reply jobs/client.go]: Done

**Review by cristianchaparroa — APPROVED** [review]

**Review by daltemen — APPROVED** [review]

**Pull request merged**